### PR TITLE
named routes, url-for

### PR DESCRIPTION
**Pull request by arohner**
Here it is.

### CODE REVIEW

File: src/noir/core.clj [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "Functions to work with partials and pages."
   (:use hiccup.core
         compojure.core)
+  (:use [clojure.contrib.except :only (throwf)])
```
Comment (ibdknox) [inline src/noir/core.clj]: Is there an alternative to using contrib here? I scrubbed contrib out of Noir to prepare for 1.3.0 and it's important for CLJS compilation.

File: src/noir/core.clj [hunk 2]
```diff
@@ -22,23 +23,60 @@
...
-                         [(keyword->symbol "compojure.core" (first rte)) (second rte)]
-                         ['compojure.core/GET rte])
-        func (symbol (route->key action url))]
-    {:action action :url url :route-fn func}))
+;; (defpage "/foo/:id" {})
+;; (defpage [:get "/foo/:id"] {})
+;; (defpage foo "/foo/:id" {})
+;; (defpage foo [:post "/foo/:id"])
+
+;; this would be a good candidate for match, once it's ready (https://github.com/swannodette/match)
```
Comment (ibdknox) [inline src/noir/core.clj:30]: I agree, I'll talk to dnolen about that tonight.

**Comment by ibdknox** [comment]
This looks great. If you can remove the contrib dep, I'll merge it in.

**Comment by arohner** [comment]
Done. I've also added more error checking to url-for.

**Pull request merged**